routers/api/v1/repo/issue.go

```diff
@@ -337,13 +337,13 @@ func EditIssue(ctx *context.APIContext, form api.EditIssueOption) {
 
 	// Update the deadline
 	var deadlineUnix timeutil.TimeStamp
-	if form.Deadline != nil && !form.Deadline.IsZero() && ctx.Repo.CanWrite(models.UnitTypeIssues) {
+	if form.Deadline != nil && ctx.Repo.CanWrite(models.UnitTypeIssues) {
 		deadlineUnix = timeutil.TimeStamp(form.Deadline.Unix())
```

Comment:
```suggestion
		deadlineUnix := timeutil.TimeStamp(form.Deadline.Unix())
```
(And remove declaration above)